docs/hf_hub.rst

```diff
@@ -0,0 +1,79 @@
+- Collaborating with others on a model
+- Convenient sharing of models with the community
+
+
+In terms of files, there are three which a scikit-learn model repo needs to
+have on the Hub:
+
+- ``README.md``: includes certain metadata on top of the file and then a
+  description of the model, aka model card.
+- ``config.json``: contains the configuration needed to run the model.
+- The persisted model file. There are no constraints on the name of the file
+  and the name is configured in ``config.json``. The file needs to be loadable
+  by ``joblib`` or ``pickle``.
+
+There are certain requirements in terms of information about the model for the
+Hub to be able to load and run the model. For scikit-learn compatible models,
+this information is stored in two places:
+
+- The metadata in ``README.md`` of the model repository, about which you can
+  read `here <https://huggingface.co/docs/hub/models-cards>`__.
+- The configuration stored in ``config.json``.
+
+As a user of ``skops``, you can use the tools in ``skops.hub_utils`` to create
+and persist a ``config.json`` file, and then use it to populate necessary
+metadata in the ``README.md`` file. The metadata in ``README.md`` is used by
+the Hub's backend to understand the type of the model and the kind of task
+which the model tries to solve. An example of a task can be
+``"tabular-classification"`` or ``"text-regression"``.
+
+An example ``config.json`` file looks like this::
+
+    {
+        "sklearn": {
+            "columns": [
+                "petal length (cm)",
+                "petal width (cm)",
+                "sepal length (cm)",
+                "sepal width (cm)",
+            ],
+            "environment": ['scikit-learn="1.1.1"', "numpy"],
+            "example_input": {
+                "petal length (cm)": [1.4, 1.4, 1.3],
+                "petal width (cm)": [0.2, 0.2, 0.2],
+                "sepal length (cm)": [5.1, 4.9, 4.7],
+                "sepal width (cm)": [3.5, 3.0, 3.2],
+            },
+            "model": {"file": "model.pkl"},
+            "task": "tabular-classification",
+        }
+    }
+
+The key ``sklearn`` includes the following sub-keys:
+
+- ``columns``: An ordered list of column names. The order is important as it is
+  used to make sure the input given to the model is what the model expects.
+- ``example_input``: A list of examples to the model. This is in the form of a
+  dictionary of column names to list of values, and is used by the Hugging Face
+  Hub backend to show them in the widget to test the model when visiting the
+  model's page on the Hub.
+- ``environment``: A list of dependencies that the model requires. These
+  packages must be available on conda-forge and are installed before loading
+  the model.
+- ``model.file``: The file name of the persisted model.
+- ``task``: The task of the model.
+
+You almost never need to create or touch this file manually, and it's created
+when you call :func:`skops.hub_utils.init`.
```

Comment:
Maybe an emphasis on this with bold font would be noice but LGTM 😅 it's one of the main value propositions of skops that's why
![image](https://user-images.githubusercontent.com/53175384/182155197-3223bd6d-32f4-4eb1-bed5-5f87ccd10f1a.png)